### ISSUE TITLE
how about return a more clearly error when http2 client try to connect some services that are not http2 server.

### ISSUE THREAD

**Issue by jingyugao** [golang/go#35724]
### What version of Go are you using (`go version`)?

1.13

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"

### What did you do?

https://play.studygolang.com/p/_1kMO-E_FEq
I use an http2 client try to connect an http client.

I expect to see an protocol error or somethig that can tell me the destination is not a valiad http2 service.

### What did you see instead?
I get an "unexpected EOF" error.
I think we can make the error more clearly.

According to  the rfc, the last segment of [https://httpwg.org/specs/rfc7540.html#rfc.section.3.5]

> Clients and servers MUST treat an invalid connection preface as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.

**Comment by bradfitz** [comment]
Yeah, I saw the associated CL earlier (which had no context) and I asked for this bug to be filed.

This doesn't affect normal net/http or x/net/http2 users. This affects people using the low-level http2.Framer directly, writing their own HTTP/2 clients.

**Comment by jingyugao** [comment]
Client base on http2 may wrap the error and return an "http2:unexpected EOF" to users. Users cant know why.If client set the `maxReadSize`, users may get and "http2: frame too large" error.
Both error cant tell user the real reason.
If we first check the frame type, we can return the exact error and we don't need to read the last content.
